src/tiled/tilesetdock.cpp

```diff
@@ -758,20 +758,31 @@ void TilesetDock::exportTileset()
     if (!tileset)
         return;
 
+    Preferences *prefs = Preferences::instance();
+
     const QLatin1String extension(".tsx");
-    QString suggestedFileName = QFileInfo(mMapDocument->fileName()).path();
-    suggestedFileName += QLatin1Char('/');
+    QLatin1String empty(".");
+    QString suggestedFileName = prefs->lastPath(Preferences::ExternalTileset);
+
+    for(int i = (suggestedFileName.length() - 1); suggestedFileName.at(i) != QLatin1Char('/')
+        && suggestedFileName != empty; i--)
+        suggestedFileName.remove(i, 1);
+
+    if(suggestedFileName == empty)
+        suggestedFileName += QLatin1Char('/');
```

Comment:
What does this do? If the suggestedFileName equals ".", it will add a slash? And what happened to the code that would suggest using the path of the map file? I think that should stay in case no path was saved yet for `ExternalTileset`.